compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/descriptors/WrappedDescriptors.kt

```diff
@@ -665,7 +672,36 @@ open class WrappedClassDescriptor(
 
     override fun getMemberScope(typeSubstitution: TypeSubstitution) = MemberScope.Empty
 
-    override fun getUnsubstitutedMemberScope() = MemberScope.Empty
+    override fun getUnsubstitutedMemberScope() = _unsubstitutedMemberScope
+
+    private val _unsubstitutedMemberScope: MemberScope = object : MemberScopeImpl() {
+        override fun printScopeStructure(p: Printer) {
+            TODO("Not yet implemented")
+        }
+
+        override fun getContributedDescriptors(
+            kindFilter: DescriptorKindFilter,
+            nameFilter: (Name) -> Boolean
+        ): Collection<DeclarationDescriptor> =
+            owner.declarations
+                .filter {
+                    // TODO: Unified way to determine _contributed_ declarations, e.g., more synthetic mark to IrDeclarationOrigin?
+                    when (it) {
+                        // @Metadata.Class has a separate place to serialize constructors.
+                        is IrConstructor -> false
+                        is IrFunction -> it.isReal && !it.origin.isSynthetic &&
+                                it.origin != IrDeclarationOrigin.ENUM_CLASS_SPECIAL_MEMBER &&
+                                // TODO: Better way to filter out <clinit>, e.g., JvmLoweredDeclarationOrigin.CLASS_STATIC_INITIALIZER?
+                                (!it.isStaticMethodOfClass || !it.nameForIrSerialization.isSpecial)
+                        is IrProperty -> !it.isFakeOverride && !it.origin.isSynthetic &&
+                                it.origin != IrDeclarationOrigin.DEFAULT_PROPERTY_ACCESSOR
+                        is IrField -> !it.isFakeOverride && !it.origin.isSynthetic &&
+                                it.origin != IrDeclarationOrigin.FIELD_FOR_OBJECT_INSTANCE &&
+                                it.origin != IrDeclarationOrigin.FIELD_FOR_ENUM_ENTRY
+                        else -> !it.isFakeOverride && !it.origin.isSynthetic
+                    }
```

Comment:
I have to admit this is quite ad-hoc, but at the same time, tweaking `synthetic` in `IrDeclarationOrigin` seems risky and/or deserves a separate commit/PR.

E.g., I found a TODO for `<clinit>`: https://github.com/JetBrains/kotlin/blob/master/compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/StaticInitializersLowering.kt#L43, since `<clinit>` is synthesized only if there are static jvm fields. But, marking it directly on `JvmLoweredDeclarationOrigin.CLASS_STATIC_INITIALIZER` makes `<clinit>` have `synthetic` modifier.

Similarly, special static fields and enum-specific members/entries are also somewhat synthetic, not user contributed/declared.